pkg/component/worker/kubelet.go

```diff
@@ -169,3 +200,22 @@ func splitRuntimeConfig(rtConfig string) (string, string, error) {
 
 	return runtimeType, runtimeSocket, nil
 }
+
+const awsMetaInformationURI = "http://169.254.169.254/latest/meta-data/local-hostname"
+
+func getNodeName() (string, error) {
+	req, err := http.NewRequest("GET", awsMetaInformationURI, nil)
```

Comment:
This ties the implementation now 100% to AWS.

Reply:
No, it has fallback to the actual os.Hostname.
The thing is that we need to have internal aws hostname if we run in ec2 and as far as I understand it is the only way to get it and check if we in the ec2.